.github/workflows/main.yaml

```diff
@@ -1,36 +1,261 @@
 #
 # Generated the prime numbers and generated used by SSH DH Group KEX.
 #
-name: GEN
+# Generating is fast. Max 15 minutes.
+# Validating is slow. 30 hours.
+#
+# Since the GitHub VM have 2 CPU each job will run 2 validating tasks.
+# A job can run for 6 hours... so 12 hours of validation time per job.
+# With 3 jobs for a certain size we should do it.
+#
+# It generates primes for 1024, 1536, 2048, 3072, 4096, 6144, 7680, 8192
+#
+name: CD
 
 # Only run on pull requests and only for Python files.
 on:
-  push:
-    branches: [ master ]
+  # Pull requests are enabled for development.
   pull_request:
-    branches: [ master ]
+    branches: [ main ]
 
+  # https://crontab.guru
+  schedule:
+    # At 02:00 on Saturday
+    - cron: '0 2 * * 6'
```

Comment:
I think that is enough if this runs once per month

Reply:
Sure.